### ISSUE TITLE
Build breakage with Kotlin `1.9.0-Beta`

### ISSUE THREAD

**Issue by sgammon** [micronaut-projects/micronaut-core#9364]
### Expected Behavior

Building with different Kotlin SDKs and language versions should not break the build

### Actual Behaviour

Instead, I get JPA errors when switching to Kotlin `1.9.0-Beta`

### Steps To Reproduce

1. Generate a Kotlin Micronaut project at latest v3
2. Upgrade to Kotlin `1.8.21`, observe that everything works
3. Upgrade to Kotlin `1.9.0-Beta`, observe breakage

### Environment Information

- Operating System: macOS Ventura 13.3.1
- JDK: GraalVM EE 22.3.1 (Java 19)

```
java version "19.0.2" 2023-01-17
Java(TM) SE Runtime Environment GraalVM EE 22.3.1 (build 19.0.2+7-jvmci-22.3-b11)
Java HotSpot(TM) 64-Bit Server VM GraalVM EE 22.3.1 (build 19.0.2+7-jvmci-22.3-b11, mixed mode, sharing)
```

### Example Application

https://github.com/sgammon/micronaut-kotlin-1-9-reproducer

### Version

3.9.2

**Comment by graemerocher** [comment]
sounds like a bug in Kotlin, have you reported them there?

**Comment by sgammon** [comment]
@graemerocher good idea, i will cross file and get back to this issue, thank you

**Comment by sgammon** [comment]
looks like this was fixed in `1.9.0-RC`
